src/pgduckdb_node.cpp

```diff
@@ -193,7 +193,11 @@ ExecuteQuery(DuckdbScanState *state) {
 		named_values[duckdb::to_string(i + 1)] = duckdb::BoundParameterData(duckdb_param);
 	}
 
-	auto pending = prepared.PendingQuery(named_values, true);
+	// Always set `allow_stream_result` to false to force a fully materialized DuckDB result.
+	// This is required for cases like CTAS from a Postgres table, where allowing streaming results
+	// could lead to race conditions on Postgres resources.
+	// Checkout discussion: https://github.com/duckdb/pg_duckdb/discussions/866
+	auto pending = prepared.PendingQuery(named_values, false);
```

Comment:
I think it would be good to only do this if postgres tables are actually involved in the query that we send to duckdb. Otherwise dataloading steps (e.g. loading a few GB parquet file) will use much more memory for no real reason, because now the result cannot be streamed anymore. The only way to really know whether a postgres query is involved in the query (given that `duckdb.query` exists) is to do this detection while preparing the statement, i.e. if it the plan involves a postgres scan, then we should not allow streaming results.

Reply:
I added a check on the rtables of the Query instance. And as long as a Postgres table is referenced, the stream is disabled.